### ISSUE TITLE
Cannot buy/sell items from merchants - Cyberpunk Red

### ISSUE THREAD

**Issue by minjaekwon9** [fantasycalendar/FoundryVTT-ItemPiles#558]
**Are you on the latest version of Item Piles?**
Yes

**Have you tried to reset Item Piles' system settings?**
Yes

**Describe the bug**
An Assistant GM created a Merchant and no other users can buy or sell items from that merchant. They are able to open the merchant's sheet and the TradeMerchantItemDialog, but the Buy Item/Sell Item buttons don't do anything.

**To Reproduce**

1. Assistant GM creates a character and changes sheet to an item pile, then merchant.
2. Populate the merchant with items. 
3. Other user attempts to trade with created merchant.

**Expected behavior**
When the Assistant GM trades with the merchant, the merchant works as expected and the button adds the item to his inventory and subtracts the correct amount of currency.

**Screenshots**
![image](https://github.com/fantasycalendar/FoundryVTT-ItemPiles/assets/71264291/4dd360e0-7da5-42fc-aa6f-118858e31a09) 
The Buy Item does not do anything unless you are the Assistant GM who created the merchant.

**Setup:**

- Foundry Version: Version 11 Stable, Build 308
- System Version: Cyberpunk RED - CORE v0.88.2
- Item Piles Version: 2.9.0
- Browser & version Google Chrome 123.0.6312.107

**Active modules:**

- Item Piles
- libwrapper
- socketlib
- various others

**Comment by Haxxer** [comment]
Is the GM logged in during the trade?

**Comment by minjaekwon9** [comment]
Yes, he runs the server from his computer so he's always logged in while I'm logged in. But for some reason only I can create merchants now. He can't convert characters into merchants anymore.

**Comment by Haxxer** [comment]
Without information about your step by step process to get the errors, screenshots, and/or console logs, I can't really help you - please provide more information.
